src/ScikitLearn/ScikitLearn.jl

```diff
@@ -37,17 +40,24 @@ function _skmodel_fit(modelname, params)
 			Xmatrix   = MLJBase.matrix(X)
 			yplain    = y
 			targnames = nothing
-			# in multi-target case
+			decode    = nothing
+			y1 		  = nothing
+			# in multi-target regression case
 			if Tables.istable(y)
 			   yplain    = MLJBase.matrix(y)
 			   targnames = MLJBase.schema(y).names
 			end
+			if eltype(y) <: CategoricalString
+				yplain = MLJBase.int(y)
+				y1     = y[1]
+			end
```

Comment:
You need  `if eltype(y) <: Union{CategoricalString,CategoricalValue}` here.  But I am wondering why you need to test  machine types at all here. Wouldn't it be safer/cleaner to have separate `_skmodel_fit_classifier` method for classifiers?

Reply:
changed to the union, thanks.

With respect to your proposal, it's a bit tricky: at macro time we don't know if it's a classifier or a regressor (we could read the  model name but that's hacky); so we could not have a `_skmodel_fit_classifier`; what we could have is that this function is duplicated something like:

```julia
function _skmodel_fit(...)
    quote
        function MLJBase.fit(model::$modelname, verbosity::Int, X, y::AbstractVector{<:Union{CategoricalString,CategoricalValue}}
         ....
        end
        function MLJBase.fit(..., X, y)
        ....
        end
    end
end
```

However this has several issues:

1. it creates (a lot of) methods which will never be used
2. it forces `y` to be an AbstractVector, I prefer `eltype` because it works on anything

So I think in this case it may be better to stick with this.

Reply:
Sorry I don't understand. Does not the caller of `@sk_model` know whether it wants a regressor or classifier? We could have `@sk_model_regressor` and `@sk_model_classifier`, no?